lightly/active_learning/utils/object_detection_output.py

```diff
@@ -126,9 +126,9 @@ def from_scores(cls,
         # create one-hot class probabilities
         max_label = max(labels) if len(labels) > 0 else 0
         class_probabilities = []
-        for label in labels:
-            c = [0.] * (max_label + 1)
-            c[label] = 1.
+        for object_prob, label in zip(object_probabilities, labels):
+            c = [(1-object_prob)/(max_label + 1)] * (max_label + 1)
+            c[label] = object_prob
```

Comment:
For this to sum to one you would need to use `max_label` instead of `max_label + 1`, or am I wrong? Can you test this somehow?

Also, I want to make you aware that the number of classes detected on an image now has an influence on the score because we evaluate the number of classes dynamically. For example, for two detections with a score of `0.7` on two different images, where there are ten classes detected on the first image and two classes on the second, the difference between the most confident class probability and the second most confident class probability will be different for the two detections.

Reply:
`max_label` can be `0`. I just rushed to get something working for all cases. I can add more tests (e.g. scores sum up to `1.0`. And yes, the scores now depend on the number of detected classes within images.

Reply:
This will make the the scorers "uncertainty_least_confidence" and "objectness_least_confidence" output exactly the same values, thus the advantages is zero.
If get the point of faking the tutorial, but having the uncertainty scores without probability vectors makes 0 sense from an active learning point of view. Thus I strongly oppose putting such fake distributions in our production code.

Reply:
1. That's not true because the number of classes varies for each image (which is a problem on its own)
2. It's not faking. If the user can't provide probabilities we need to do the best we can.
3. We need to think about these kinds of problems before we implement major changes. Now we only have two options. Either we merge this fix and release today or we postpone the release and implement all the infrastructure need so that we can go past using default scores and have the user decide which score to use.

Reply:
This is not about "making it work" but about "faking it to make it seem as if works" which should not be part of the production code.

Reply:
This is not about faking or not. It's about allowing the user to use our AL workflow when has no class probabilities but only labels. We have to be transparent and tell the user that the scores are not as expected when using only labels and not class probabilities. I would also add to the documentation that we strongly advise against using the `.from_scores(..)` method. 
If possible users should provide class probabilities. But at the same time, we want to make sure, all users can benefit from AL without a lot of extra work.